types/src/data_availability_header.rs

```diff
@@ -53,22 +53,32 @@ use crate::{bail_validation, Error, ExtendedDataSquare, Result, ValidateBasic, V
 )]
 pub struct DataAvailabilityHeader {
     /// Merkle roots of the [`ExtendedDataSquare`] rows.
-    pub row_roots: Vec<NamespacedHash>,
+    row_roots: Vec<NamespacedHash>,
     /// Merkle roots of the [`ExtendedDataSquare`] columns.
-    pub column_roots: Vec<NamespacedHash>,
+    column_roots: Vec<NamespacedHash>,
 }
 
 impl DataAvailabilityHeader {
+    /// Create new [`DataAvailabilityHeader`].
+    ///
+    /// [`DataAvailabilityHeader::validate_basic`] should be called after this to check validity.
+    pub fn new(row_roots: Vec<NamespacedHash>, column_roots: Vec<NamespacedHash>) -> Self {
```

Comment:
if this can create invalid DAH, maybe `new_unchecked`?

Reply:
Or make this new checked and add an extension trait with unchecked version for testing, but that feels like overkill.

Reply:
I'm ok on having both actually.

Reply:
Added both, what do you think?